light/errors.go

```diff
@@ -20,6 +53,134 @@ func (e ErrOldHeaderExpired) Error() string {
 	return fmt.Sprintf("old header has expired at %v (now: %v)", e.At, e.Now)
 }
 
+type ErrTargetBlockHeightLessThanTrusted struct {
+	Target  int64
+	Trusted int64
+}
+
+func (e ErrTargetBlockHeightLessThanTrusted) Error() string {
+	return fmt.Sprintf("target block has a height lower than the trusted height (%d < %d)", e.Target, e.Trusted)
+}
+
+type ErrHeaderHeightNotMonotonic struct {
+	WantHeight int64
+	GetHeight  int64
+}
+
+func (e ErrHeaderHeightNotMonotonic) Error() string {
+	return fmt.Sprintf("expected new header height %d to be greater than one of old header %d", e.WantHeight, e.GetHeight)
+}
+
+type ErrHeaderTimeNotMonotonic struct {
+	WantTime time.Time
+	GetTime  time.Time
+}
+
+func (e ErrHeaderTimeNotMonotonic) Error() string {
+	return fmt.Sprintf("expected new header time %v to be after old header time %v", e.WantTime, e.GetTime)
+}
+
+type ErrHeaderTimeExceedMaxClockDrift struct {
+	Ti    time.Time
+	Now   time.Time
+	Drift time.Duration
+}
+
+func (e ErrHeaderTimeExceedMaxClockDrift) Error() string {
+	return fmt.Sprintf("new header has a time from the future %v (now: %v; max clock drift: %v)", e.Ti, e.Now, e.Drift)
+}
+
+type ErrUnverifiedHeight struct {
+	Height int64
+}
+
+func (e ErrUnverifiedHeight) Error() string {
+	return fmt.Sprintf("unverified header/valset requested (latest: %d)", e.Height)
+}
+
+type ErrInvalidTrustLevel struct {
+	Level cmtmath.Fraction
+}
+
+func (e ErrInvalidTrustLevel) Error() string {
+	return fmt.Sprintf("trustLevel must be within [1/3, 1], given %v", e.Level)
+}
+
+type ErrValidatorsMismatch struct {
+	HeaderHash     cmtbytes.HexBytes
+	ValidatorsHash []byte
```

Comment:
Why are these fields of different types?
The expressions in the calling code probably are, but this is public API, let's do a more consistent job at representing hash data.